### PR TITLE
Make TransitiveLicensesInfo public + alternative

### LINKED ISSUE #86: TransitiveLicensesInfo was made private, breaking downstream customers
https://github.com/bazelbuild/rules_license/commit/88789cce44d2123f2814cea3c52c87b4d047d05b made TransitiveLicensesInfo private, but that is breaking downstream users of gather_licenses_info.

We need to do that in a soft transition and provide an alternative.

### PR DESCRIPTION
**Pull request by junglegoog**
Fixes #86

1. The private TransitiveLicensesInfo is symlinked as (deprecated) public provider.
2. write_licenses_info returns the File objects referenced by the licenses json file. This return value reduces the need to depend on TransitiveLicensesInfo publicly.

- [x] Tests pass
- [ ] Tests and examples for any new features.
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

File: rules/gather_licenses_info.bzl [hunk 1]
```diff
@@ -30,9 +30,9 @@ def _strip_null_repo(label):
     The is to make str(label) compatible between bazel 5.x and 6.x
     """
     s = str(label)
-    if s.startswith('@//'):
+    if s.startswith("@//"):
```
Comment (aiuto) [inline rules/gather_licenses_info.bzl:33]: Don't change what is not part of the intent. That messes up the blame layer.

File: rules/gather_licenses_info.bzl [hunk 2]
```diff
@@ -135,15 +135,25 @@ def write_licenses_info(ctx, deps, json_out):
...
+      A list of License File objects for each of the license paths referenced in the json.
     """
-    licenses = []
+    licenses_json = []
+    licenses_files = []
     for dep in deps:
         if TransitiveLicensesInfo in dep:
-            licenses.extend(licenses_info_to_json(dep[TransitiveLicensesInfo]))
+            transitive_licenses_info = dep[TransitiveLicensesInfo]
+            licenses_json.extend(licenses_info_to_json(transitive_licenses_info))
```
Comment (aiuto) [inline rules/gather_licenses_info.bzl:147]: This seems to be from a different PR.

File: rules/providers.bzl [hunk 3]
```diff
@@ -63,5 +68,9 @@ ExperimentalMetadataInfo = provider(
         "type": "string: How to interpret data",
         "label": "Label: label of the metadata rule",
         "data": "String->any: Map of names to values",
-    }
+    },
 )
+
+# Deprecated: Use write_licenses_info instead.
+# TODO: Remove once downstream customers are migrated off TransitiveLicensesInfo .
```
Comment (aiuto) [inline rules/providers.bzl:75]: TODO(version 0.0.6): Drop this.

We can't make an indefinite promise that people have migrated off.
At the 0.0.x level, we are allowed breaking changes at any release.
The 0.1.0 is certainly not going to try to be fully backwards compatible with 0.0.whatever

**Review by aiuto — CHANGES_REQUESTED** [review]
I think this change should only be touching providers.bzl.

File: rules/gather_licenses_info.bzl (hunk 2 above)
Comment (junglegoog) [reply rules/gather_licenses_info.bzl:147]: No, it is this PR. The Files returned from licenses_info_to_json are added to licenses_json which is returned by write_licenses_info.

Comment (junglegoog) [reply rules/gather_licenses_info.bzl:147]: This is needed, so that downstream customers have an alternative to using TransitiveLicensesInfo directly, so it can be deprecated.

File: rules/providers.bzl (hunk 3 above)
Comment (junglegoog) [reply rules/providers.bzl]: Dropped.

**Comment by junglegoog** [comment]
I answered / resolved the comments. PTAL again.

**Comment by junglegoog** [comment]
I updated the PR description, and also the usage instructions in write_licenses_info pydoc.

File: rules/gather_licenses_info.bzl (hunk 2 above)
Comment (aiuto) [reply rules/gather_licenses_info.bzl:153]: Well... It's not going to matter much.   gather_licenses_info is going to be deprecated too over time.
Google will use it internally for a while, but I'll b doing all examples around user PacakgeInfo and user defined providers.

**Review by aiuto — APPROVED** [review]
